Documentation/Functions/Typolink.rst

```diff
@@ -543,27 +543,51 @@ userFunc
     :name: typolink-userFunc
     :type: :ref:`data-type-function-name`
 
-    This passes the link-data compiled by the typolink function to a user-
-    defined function for final manipulation.
-
-    The :php:`$content` variable passed to the user-function (first parameter) is
-    an array with the keys "TYPE", "TAG", "url", "targetParams" and
-    "aTagParams".
-
-    TYPE is an indication of link-kind: mailto, url, file, page
-
-    TAG is the full <A>-tag as generated and ready from the typolink
-    function.
-
-    The actual tag value is constructed like this:
-
-    ..  code-block:: php
-
-        $contents = '<a href="' . $finalTagParts['url'] . '"'
-                   . $finalTagParts['targetParams']
-                   . $finalTagParts['aTagParams'] . '>';
-
-    The userfunction must return an <A>-tag.
+    All of the :typoscript:`typolink` TypoScript configuration will be parsed and evaluated
+    by the TYPO3 Core's :php:`LinkFactory->create()` method, and then passed on
+    to the defined :typoscript:`userFunc` for further manipulation. The :typoscript:`userFunc` needs to
+    return an object implementing the :php:`LinkResultInterface`. The currently calculated
+    typolink is passed as an argument to the :php:`userFunc` as an object of the same type. This allows
+    to return either an enriched link, or a completely new one.
+
+    The detailed execution steps are:
+
+    First, the :typoscript:`typolink` will be created as configured by the specified TypoScript.
+    This will result in an object of Type :php:`LinkResultInterface`. This object receives
+    all of the TypoScript :typoscript:`typolink` configuration as properties, and makes them
+    available via corresponding getters and setters.
+    Then your custom :typoscript:`userFunc` is executed and receives the following arguments
+    (delivered via :php:`$contentObjectRenderer->callUserFunction()`):
+
+    `$content`
+        This contains the object implementing :php:`LinkResultInterface`. Inside your :php:`userFunc()` you
+        can call for example:
+
+        *  :php:`$content->getUrl()` to get the URL of a link,
+        *  :php:`$content->getLinkText()` to get the title of your link,
```

Comment:
This is not the link title. This is the text in between: `<a href="https://obi.de">OBI-wan KenOBI is the link text</a>`

Reply:
Reworded this.